core/src/main/java/com/graphhopper/routing/weighting/custom/PriorityCalculator.java

```diff
@@ -24,112 +24,53 @@
 import org.locationtech.jts.geom.prep.PreparedGeometryFactory;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 
 final class PriorityCalculator {
     private final List<EdgeToValueEntry> priorityList = new ArrayList<>();
 
     public PriorityCalculator(CustomModel customModel, EncodedValueLookup lookup) {
-        add(lookup, customModel.getVehicleWeight(), "vehicle_weight", MaxWeight.KEY);
-        add(lookup, customModel.getVehicleWidth(), "vehicle_width", MaxWidth.KEY);
-        add(lookup, customModel.getVehicleHeight(), "vehicle_height", MaxHeight.KEY);
-        add(lookup, customModel.getVehicleLength(), "vehicle_length", MaxLength.KEY);
-
         for (Map.Entry<String, Object> entry : customModel.getPriority().entrySet()) {
             String key = entry.getKey();
+            String priorityKey = "priority." + key;
             Object value = entry.getValue();
-
-            if (value == null) {
+            if (value == null)
                 throw new IllegalArgumentException("Missing value for " + key + " in 'priority'");
-            } else if (value instanceof Number) {
-                if (key.startsWith(GeoToValueEntry.AREA_PREFIX)) {
-                    Geometry geometry = GeoToValueEntry.pickGeometry(customModel, key);
-                    priorityList.add(new GeoToValueEntry(new PreparedGeometryFactory().create(geometry), ((Number) value).doubleValue(), 1));
+
+            if (key.startsWith(GeoToValueEntry.AREA_PREFIX)) {
+                if (!(value instanceof Number))
+                    throw new IllegalArgumentException(priorityKey + ": area entry requires number value but was: " + value.getClass().getSimpleName());
+                Geometry geometry = GeoToValueEntry.pickGeometry(customModel, key);
+                priorityList.add(GeoToValueEntry.create(priorityKey, new PreparedGeometryFactory().create(geometry),
+                        (Number) value, 1, 0, 1));
```

Comment:
maybe use `final double defaultPriority=1; final double minPriority=0; final double maxPriority=1;` for readability and to indicate its always the same